src/hashtable.c

```diff
@@ -79,7 +79,7 @@ static pair_t *hashtable_find_pair(hashtable_t *hashtable, bucket_t *bucket,
     list = bucket->first;
     while (1) {
         pair = list_to_pair(list);
-        if (pair->hash == hash && strcmp(pair->key, key) == 0)
+        if (pair->hash == hash && strncmp(pair->key, key, key_len) == 0)
```

Comment:
`strncmp()` doesn't work exactly right for this: if `key` is a prefix of `pair->key` then this will think they match. You need to compare the length of the `pair->key` to `key_len` first. And if you want to support a `'\0'` in the middle of a key string (which `json_stringn()` supports), then you need to store the key length explicitly in every `pair_t`.

Reply:
Fixed, thanks